cmd/otel-agent/command/command.go

```diff
@@ -59,16 +62,36 @@ func makeCommands(globalParams *subcommands.GlobalParams) *cobra.Command {
 	flagSet := flags(featuregate.GlobalRegistry(), globalParams)
 	otelAgentCmd.PersistentFlags().AddGoFlagSet(flagSet)
 
+	// Support these environment variables
+	ef := envflag.NewEnvFlag(flagSet, 2,
+		map[string]string{ // User-defined env-flag map
+			"DD_SYNC_DELAY":  "sync-delay",
+			"DD_SYNC_TO":     "sync-to",
+			"DD_CORE_CONFIG": "core-config",
+		},
+		true, // show env variable key in usage
+		true, // show env variable value in usage
+	)
+
+	if err := ef.Parse(os.Args[1:]); err != nil {
+		panic(err)
+	}
+
 	return &otelAgentCmd
 }
 
 const configFlag = "config"
+const coreConfigFlag = "core-config"
+const syncDelayFlag = "sync-delay"
+const syncTimeoutFlag = "sync-to"
 
 func flags(reg *featuregate.Registry, cfgs *subcommands.GlobalParams) *flag.FlagSet {
 	flagSet := new(flag.FlagSet)
-
 	flagSet.Var(cfgs, configFlag, "Locations to the config file(s), note that only a"+
 		" single location can be set per flag entry e.g. `--config=file:/path/to/first --config=file:path/to/second`.")
+	flagSet.StringVar(&cfgs.CoreConfPath, coreConfigFlag, "", "Location to the Datadog Agent config file.")
+	flagSet.DurationVar(&cfgs.SyncDelay, syncDelayFlag, 0, "Delay before first config sync.")
+	flagSet.DurationVar(&cfgs.SyncTimeout, syncTimeoutFlag, 3*time.Second, "Timeout for sync requests.")
```

Comment:
can we remove this ?

Reply:
Can't we keep the CLI flags even if they go unused? I'd have to make larger changes to get rid of this. We now have env vars to assign these same vars. If you *really* want to get rid of it, can I do it in a follow-up cleanup PR?